testnet mutation error

```
** (Mix) Could not start application ae_mdw: bad return value: {{AeMdw.Application, :start_phase, [:migrate_db, :normal, []]}, {:EXIT, {{:badmatch, {:aborted, {%Protocol.UndefinedError{description: "", protocol: AeMdw.Db.Mutation, value: []}, [{AeMdw.Db.Mutation, :impl_for!, 1, [file: 'lib/ae_mdw/db/mutation.ex', line: 1]}, {AeMdw.Db.Mutation, :mutate, 1, [file: 'lib/ae_mdw/db/mutation.ex', line: 16]}, {Enum, :"-each/2-lists^foreach/1-0-", 2, [file: 'lib/enum.ex', line: 783]}, {Enum, :each, 2, [file: 'lib/enum.ex', line: 783]}, {:mnesia_tm, :apply_fun, 3, [file: 'mnesia_tm.erl', line: 842]}, {:mnesia_tm, :execute_transaction, 5, [file: 'mnesia_tm.erl', line: 818]}, {AeMdw.Mnesia, :transaction, 1, [file: 'lib/ae_mdw/mnesia.ex', line: 161]}, {AeMdw.Migrations.OracleRespondInternalCalls, :run, 1, [file: 'priv/migrations/20220124162300_oracle_respond_internal_calls.ex', line: 40]}, {Mix.Tasks.MigrateDb, :apply_migration!, 2, [file: 'lib/mix/tasks/migrate_db.ex', line: 87]}, {Enum, :"-map/2-lists^map/1-0-", 2, [file: 'lib/enum.ex', line: 1396]}, {Mix.Tasks.MigrateDb, :run, 1, [file: 'lib/mix/tasks/migrate_db.ex', line: 40]}, {AeMdw.Application, :start_phase, 3, [file: 'lib/ae_mdw/application.ex', line: 238]}, {:application_starter, :run_the_phase, 4, [file: 'application_starter.erl', line: 99]}, {:application_starter, :start_apps, 3, [file: 'application_starter.erl', line: 59]}, {:application_starter, :start, 3, [file: 'application_starter.erl', line: 45]}, {:application_master, :start_the_app, 5, [file: 'application_master.erl', line: 316]}, {:application_master, :start_it_new, 7, [file: 'application_master.erl', line: 299]}]}}}, [{AeMdw.Mnesia, :transaction, 1, [file: 'lib/ae_mdw/mnesia.ex', line: 160]}, {AeMdw.Migrations.OracleRespondInternalCalls, :run, 1, [file: 'priv/migrations/20220124162300_oracle_respond_internal_calls.ex', line: 40]}, {Mix.Tasks.MigrateDb, :apply_migration!, 2, [file: 'lib/mix/tasks/migrate_db.ex', line: 87]}, {Enum, :"-map/2-lists^map/1-0-", 2, [file: 'lib/enum.ex', line: 1396]}, {Mix.Tasks.MigrateDb, :run, 1, [file: 'lib/mix/tasks/migrate_db.ex', line: 40]}, {AeMdw.Application, :start_phase, 3, [file: 'lib/ae_mdw/application.ex', line: 238]}, {:application_starter, :run_the_phase, 4, [file: 'application_starter.erl', line: 99]}, {:application_starter, :start_apps, 3, [file: 'application_starter.erl', line: 59]}, {:application_starter, :start, 3, [file: 'application_starter.erl', line: 45]}, {:application_master, :start_the_app, 5, [file: 'application_master.erl', line: 316]}, {:application_master, :start_it_new, 7, [file: 'application_master.erl', line: 299]}]}}}
make: *** [Makefile:37: shell] Segmentation fault (core dumped)
```

Reproducible with testnet .art database snapshot.

--- a/lib/ae_mdw/db/sync/transaction.ex
+++ b/lib/ae_mdw/db/sync/transaction.ex
@@ -131,7 +131,6 @@ defmodule AeMdw.Db.Sync.Transaction do
       WriteFieldsMutation.new(type, tx, block_index, txi),
       inner_tx_mutations
     ]
-    |> Enum.reject(&is_nil/1)
     |> List.flatten()
   end
 
@@ -224,12 +223,11 @@ defmodule AeMdw.Db.Sync.Transaction do
       |> Enum.with_index(txi)
       |> Enum.flat_map(&transaction_mutations(&1, tx_ctx))
 
-    mutations =
-      List.flatten([
-        MnesiaWriteMutation.new(Model.Block, mb_model),
-        txs_mutations,
-        Aex9AccountPresenceMutation.new(height, mbi)
-      ])
+    mutations = [
+      MnesiaWriteMutation.new(Model.Block, mb_model),
+      txs_mutations,
+      Aex9AccountPresenceMutation.new(height, mbi)
+    ]
 
     {mutations, {txi + length(mb_txs), mbi + 1}}
   end

--- a/lib/ae_mdw/mnesia.ex
+++ b/lib/ae_mdw/mnesia.ex
@@ -157,12 +157,14 @@ defmodule AeMdw.Mnesia do
 
   @spec transaction([Mutation.t()]) :: :ok
   def transaction(mutations) do
+    mutations =
+      mutations
+      |> List.flatten()
+      |> Enum.reject(&is_nil/1)
+
     {:atomic, :ok} =
       :mnesia.sync_transaction(fn ->
-        mutations
-        |> List.flatten()
-        |> Enum.reject(&is_nil/1)
-        |> Enum.each(&Mutation.mutate/1)
+        Enum.each(mutations, &Mutation.mutate/1)
       end)
 
     :ok

--- a/lib/ae_mdw/mnesia.ex
+++ b/lib/ae_mdw/mnesia.ex
@@ -159,7 +159,10 @@ defmodule AeMdw.Mnesia do
   def transaction(mutations) do
     {:atomic, :ok} =
       :mnesia.sync_transaction(fn ->
-        Enum.each(mutations, &Mutation.mutate(&1))
+        mutations
+        |> List.flatten()
+        |> Enum.reject(&is_nil/1)
+        |> Enum.each(&Mutation.mutate/1)
       end)
 
     :ok

--- a/priv/migrations/20211220163202_index_ga_attach_txs.ex
+++ b/priv/migrations/20211220163202_index_ga_attach_txs.ex
@@ -108,9 +108,7 @@ defmodule AeMdw.Migrations.IndexGaAttachTxs do
         end)
       end)
 
-    txs_mutations
-    |> List.flatten()
-    |> Mnesia.transaction()
+    Mnesia.transaction(txs_mutations)
 
     new_contracts_count = length(txs_mutations)
 

--- a/test/ae_mdw/db/sync/transaction_test.exs
+++ b/test/ae_mdw/db/sync/transaction_test.exs
@@ -39,7 +39,9 @@ defmodule AeMdw.Db.Sync.TransactionTest do
           )
 
         fn ->
-          Enum.each(mutations, &Mutation.mutate/1)
+          mutations
+          |> Enum.reject(&is_nil/1)
+          |> Enum.each(&Mutation.mutate/1)
 
           {sender_pk, recipient_pk} = pubkeys_from_tx(signed_tx)
           assert sender_pk != recipient_pk
@@ -75,7 +77,9 @@ defmodule AeMdw.Db.Sync.TransactionTest do
           )
 
         fn ->
-          Enum.each(mutations, &Mutation.mutate/1)
+          mutations
+          |> Enum.reject(&is_nil/1)
+          |> Enum.each(&Mutation.mutate/1)
 
           {sender_pk, recipient_pk} = pubkeys_from_tx(signed_tx)
           assert sender_pk == recipient_pk

